### ISSUE TITLE
Update GimmeRoQ.sh with stable build

### ISSUE THREAD

**Issue by cduverne** [roq-messaging/RoQ#121]
Update the installation script to allow users to choose between latest CI Dev or Stable.

**Comment by cduverne** [comment]
#119

**Comment by cduverne** [comment]
Hello,

I'm getting a GPG question when launching the mvn install after updating with ZMQ 3.X, JZMQ 2.1 etc... (which are correctly fetched) : 

[INFO] Building jar: /var/lib/RoQ/jzmq/target/jzmq-2.1.0-javadoc.jar
[INFO] [gpg:sign {execution: sign-artifacts}]

After this the script is hanging waiting for "GPG Passphrase"

What's the matter ?
Thanks
Cyrille

**Comment by cduverne** [comment]
Here is the command leading to this : 

#Install Maven JZMQ
mvn install -e -DskipTests >> $INSTALLDIR/roq.log

**Comment by sskhiri** [comment]
yes indeed I got the same. It seems that is used for release management of the JZMQ team. The only way I manage to install it is to to comment the involved plugin in the maven pom file of JZMQ. 
Anyway the mvn install is not required anymore since the JZMQ 2.2 that we use is dowloaded directly from the central maven repository, while the local mvn install will only install the 2.3-SNAPSHOT which is not required.
So in summary,
1. We do still need to get JZMQ to compile and installing JZMQ with the make install
2. We do NOT need the mvn install anymore since it will be downloaded from the central repo

I hope it helps !

**Comment by cduverne** [comment]
Hello,

This had been solved by removing the mvn install.
I didn't notice that the URL of the build changed, I updated them in
the script.

Let's finish it.
I'll commit tonight.

Cheers
CyD

 "Success is getting what you want. Happiness is wanting what you get"
Dale Carnegie

At Sunday, 24/03/2013 on 20:57 Sabri Skhiri wrote:

yes indeed I got the same. It seems that is used for release
management of the JZMQ team. The only way I manage to install it is to
to comment the involved plugin in the maven pom file of JZMQ. 
Anyway the mvn install is not required anymore since the JZMQ 2.2 that
we use is dowloaded directly from the central maven repository, while
the local mvn install will only install the 2.3-SNAPSHOT which is not
required.
So in summary,
1. We do still need to get JZMQ to compile and installing JZMQ with
the make install
2. We do NOT need the mvn install anymore since it will be downloaded
from the central repo

I hope it helps !

—
Reply to this email directly or view it on GitHub [1].

## Links:

[1]
https://github.com/roq-messaging/RoQ/issues/121#issuecomment-15367372

**Comment by cduverne** [comment]
Not that easy finally,

Getting an error when installing the back-end :

"Build failed:  -> task failed (err #1): {task: cxx binding.cc ->
binding_1.o}
npm ERR! error installing zmq@2.0.3
npm http GET https://registry.npmjs.org/glob
npm http GET https://registry.npmjs.org/mkdirp
npm http GET https://registry.npmjs.org/runforcover
npm http GET https://registry.npmjs.org/nopt
npm http GET https://registry.npmjs.org/slide
npm http GET https://registry.npmjs.org/difflet
npm http GET https://registry.npmjs.org/deep-equal
npm http GET https://registry.npmjs.org/buffer-equal

npm ERR! zmq@2.0.3 preinstall: `node-waf clean || (exit 0); node-waf
configure build`
npm ERR! `sh "-c" "node-waf clean || (exit 0); node-waf configure
build"` failed with 1
npm ERR! 
npm ERR! Failed at the zmq@2.0.3 preinstall script.
npm ERR! This is most likely a problem with the zmq package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node-waf clean || (exit 0); node-waf configure build
npm ERR! You can get their info via:
npm ERR!     npm owner ls zmq
npm ERR! There is likely additional logging output above.
npm ERR! 
npm ERR! System Linux 3.2.0-39-generic
npm ERR! command "node" "/usr/bin/npm" "install"
npm ERR! cwd /var/lib/RoQ/RoQ/roq-backend
npm ERR! node -v v0.6.12
npm ERR! npm -v 1.1.4
npm ERR! code ELIFECYCLE
npm ERR! message zmq@2.0.3 preinstall: `node-waf clean || (exit 0);
node-waf configure build`
npm ERR! message `sh "-c" "node-waf clean || (exit 0); node-waf
configure build"` failed with 1
npm ERR! errno {}
"

I think that the back-end doesn't need to be installed at each run of
GimmeRoQ, then I'll add a parameter to the launching command.

This done, I'll commit, but I think that the script isn't a good way
to deploy RoQ anymore, a clean package, or a chef/puppet cookbook
might be of use (but need the customers to have chef or puppet), the
package could be an interesting way.

Cheers
Cyrille

 "Success is getting what you want. Happiness is wanting what you get"
Dale Carnegie

At Sunday, 24/03/2013 on 20:57 Sabri Skhiri wrote:

yes indeed I got the same. It seems that is used for release
management of the JZMQ team. The only way I manage to install it is to
to comment the involved plugin in the maven pom file of JZMQ. 
Anyway the mvn install is not required anymore since the JZMQ 2.2 that
we use is dowloaded directly from the central maven repository, while
the local mvn install will only install the 2.3-SNAPSHOT which is not
required.
So in summary,
1. We do still need to get JZMQ to compile and installing JZMQ with
the make install
2. We do NOT need the mvn install anymore since it will be downloaded
from the central repo

I hope it helps !

—
Reply to this email directly or view it on GitHub [1].

## Links:

[1]
https://github.com/roq-messaging/RoQ/issues/121#issuecomment-15367372

**Comment by cduverne** [comment]
And the commit has been done.

Seems working for me, but the problem with the back-end has just been
avoided, and need to be solved in another issue.

Cheers
Cyrille

 "Success is getting what you want. Happiness is wanting what you get"
Dale Carnegie

At Sunday, 24/03/2013 on 20:57 Sabri Skhiri wrote:

yes indeed I got the same. It seems that is used for release
management of the JZMQ team. The only way I manage to install it is to
to comment the involved plugin in the maven pom file of JZMQ. 
Anyway the mvn install is not required anymore since the JZMQ 2.2 that
we use is dowloaded directly from the central maven repository, while
the local mvn install will only install the 2.3-SNAPSHOT which is not
required.
So in summary,
1. We do still need to get JZMQ to compile and installing JZMQ with
the make install
2. We do NOT need the mvn install anymore since it will be downloaded
from the central repo

I hope it helps !

—
Reply to this email directly or view it on GitHub [1].

## Links:

[1]
https://github.com/roq-messaging/RoQ/issues/121#issuecomment-15367372

**Comment by sskhiri** [comment]
HI on the develop branch, the GimmeRoQ.sh still shows this line:
sudo apt-get install -y libzmq-dbg=2.2.0+dfsg-2ubuntu1 libzmq-dev=2.2.0+dfsg-2ubuntu1 libzmq1=2.2.0+dfsg-2ubuntu
does it mean that we still install ZMS 2.2.0 ?

**Comment by cduverne** [comment]
Damned...

We shouldn't have this anymore, I'll update it today.

Cheers
CyD

Envoyé depuis un mobile Samsung

-------- Message d'origine --------
De : Sabri Skhiri  
Date : 
A : roq-messaging/RoQ  
Cc : Cyrille Duverne  
Objet : Re: [RoQ] Update GimmeRoQ.sh with stable build (#121) 

HI on the develop branch, the GimmeRoQ.sh still shows this line:
sudo apt-get install -y libzmq-dbg=2.2.0+dfsg-2ubuntu1
libzmq-dev=2.2.0+dfsg-2ubuntu1 libzmq1=2.2.0+dfsg-2ubuntu
does it mean that we still install ZMS 2.2.0 ?

---

**Comment by cduverne** [comment]
The version of ZMQ has been updated.

Hope it's OK now.

Enjoy
CyD

 "Success is getting what you want. Happiness is wanting what you get"
Dale Carnegie

At Sunday, 24/03/2013 on 20:57 Sabri Skhiri wrote:

yes indeed I got the same. It seems that is used for release
management of the JZMQ team. The only way I manage to install it is to
to comment the involved plugin in the maven pom file of JZMQ. 
Anyway the mvn install is not required anymore since the JZMQ 2.2 that
we use is dowloaded directly from the central maven repository, while
the local mvn install will only install the 2.3-SNAPSHOT which is not
required.
So in summary,
1. We do still need to get JZMQ to compile and installing JZMQ with
the make install
2. We do NOT need the mvn install anymore since it will be downloaded
from the central repo

I hope it helps !

—
Reply to this email directly or view it on GitHub [1].

## Links:

[1]
https://github.com/roq-messaging/RoQ/issues/121#issuecomment-15367372

**Comment by cduverne** [comment]
Validated by @sskhiri !! Therefore closed
